src/local/room_position/pair_utils.rs

```diff
@@ -5,28 +5,30 @@ impl Position {
     /// Returns this position's in-room coordinates as a pair of unsigned
     /// integers.
     #[inline]
-    pub fn coords(&self) -> (u32, u32) {
+    pub fn coords(&self) -> (u8, u8) {
         (self.x(), self.y())
     }
 
     /// Returns this position's in-room coordinates as a pair of signed
     /// integers.
     #[inline]
-    pub fn coords_signed(&self) -> (i32, i32) {
-        (self.x() as i32, self.y() as i32)
+    pub fn coords_signed(&self) -> (i8, i8) {
+        (self.x() as i8, self.y() as i8)
     }
 }
 
 // Note: we would usually implement `From<Position> for (u8, u8)`, but this
 // implementation is not allowed as it'd be implementing it on a nested type,
 // and both the outer type (`(T, T)`), and the inner ones (`u8`) are from an
 // external crate.
+// TODO: We can do stuff like impl From<LocalCostMatrix> for Vec<u8>, so maybe
+// we can do the same here for more idiomatic code?
 
 impl Into<(u8, u8)> for Position {
     /// Turns this positions's in-room coordinates into a pair of `u8` values.
     #[inline]
     fn into(self) -> (u8, u8) {
```

Comment:
Can you call self.coords()? Is it worth having an 'into' for i8 as well?

Reply:
I'm actually somewhat against having Into<(u8, u8)> and related at all, since it's ambiguous whether it's meant to be world coordinates or in-room. But it could use self.coords() here, just more explicit that this is the in-room when written like this.

Reply:
Agreed on the ambiguity so that's fine to ignore. Using coords just saves some type casting and is a bit cleaner but it's a minor nitpick.